File: WorkcraftCore/src/main/java/org/workcraft/dom/ModelDescriptor.java
package org.workcraft.dom;

import org.workcraft.dependencymanager.advanced.user.StorageManager;
import org.workcraft.interop.GlobalService;
import org.workcraft.interop.ModelService;
import org.workcraft.interop.ModelServices;

public interface ModelDescriptor {
	GlobalService<ModelDescriptor> GLOBAL_SERVICE_HANDLE = GlobalService.createNewService(ModelDescriptor.class, "Model descriptor");
	ModelService<ModelDescriptor> SERVICE_HANDLE = ModelService.createNewService(ModelDescriptor.class, "Model descriptor");
	String getDisplayName();
	/**
	 * This method returns a ServiceProvider object representing all the services the model is capable of providing.
	 */
	ModelServices newDocument();
	/**
	 * The method is called by the 'load' mechanism. This should be eliminated somehow.
	 * @param model
	 * @param storage
	 * @return
	 */
	@Deprecated
	ModelServices createServiceProvider(Model model, StorageManager storage);
}

File: PetriNetPlugin21/src/main/scala/org/workcraft/plugins/petri21/ModelDescriptor.scala
package org.workcraft.plugins.petri21
import org.workcraft.dom.ModelDescriptor
import org.workcraft.interop.ModelServices
import org.workcraft.interop.ModelServicesImpl
import org.workcraft.dependencymanager.advanced.user.StorageManager
import org.workcraft.dom.Model
import org.workcraft.exceptions.NotImplementedException

class PetriNetModelDescriptor extends ModelDescriptor {
  def getDisplayName = "Petri net"
  def newDocument = ModelServicesImpl.EMPTY
  def createServiceProvider (model: Model, storage: StorageManager) = 
    throw new NotImplementedException //ModelServicesImpl.EMPTY plus
}